JRuby Jar comuptation fails in OSGi environment

**Environment Information**
  
Provide at least:
* JRuby version: 9.4.7.0
* Operating system and platform: Windows 10 64 Bit, OSGi with Equinox

**Expected Behavior**

* JRuby should run in OSGi environment
* Any kind of access to the classpath need to take OSGi into consideration

**Actual Behavior**

* [ClasspathLauncher](https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/util/ClasspathLauncher.java#L37) expects a JAR classpath entry. In OSGi case this is a bundleresource (e.g. bundleresource://344.fwk257265217/org/jruby/Main.class).
* Path.of fails with java.nio.file.FileSystemNotFoundException: Provider "bundleresource" not installed
* Catching this exception and return null would be sufficient in our scenario.

It seems I am running into this same issue when I run a compiled JRuby app as an Atlassian Jira OSGi plugin (Jruby 9.4.7.0, MacOS). Here is the stacktrace:
```
Caused by: java.nio.file.FileSystemNotFoundException: Provider "bundle" not installed
	at java.base/java.nio.file.Path.of(Path.java:212)
	at java.base/java.nio.file.Paths.get(Paths.java:97)
	at org.jruby.util.ClasspathLauncher.getJRubyJar(ClasspathLauncher.java:40)
	at org.jruby.util.ClasspathLauncher.jrubyCommand(ClasspathLauncher.java:48)
	at org.jruby.RubyInstanceConfig.getEnvironment(RubyInstanceConfig.java:679)
	at org.jruby.util.OSEnvironment.environmentVariableMap(OSEnvironment.java:51)
	at org.jruby.RubyGlobal.defineGlobalEnvConstants(RubyGlobal.java:395)
	at org.jruby.RubyGlobal.createGlobals(RubyGlobal.java:234)
	at org.jruby.Ruby.<init>(Ruby.java:520)
	at org.jruby.Ruby.newInstance(Ruby.java:723)
```
And as @glatuske mentioned, just fetching the `FileSystemNotFoundException` would be enough to proceed.

--- a/core/src/main/java/org/jruby/util/ClasspathLauncher.java
+++ b/core/src/main/java/org/jruby/util/ClasspathLauncher.java
@@ -5,6 +5,7 @@ import java.net.MalformedURLException;
 import java.net.URISyntaxException;
 import java.net.URL;
 import java.net.URLClassLoader;
+import java.nio.file.FileSystemNotFoundException;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
@@ -38,7 +39,7 @@ public class ClasspathLauncher {
 
         try {
             return Paths.get(new URL(jarPath).toURI()).toString();
-        } catch (URISyntaxException | MalformedURLException e) {
+        } catch (URISyntaxException | MalformedURLException | FileSystemNotFoundException e) {
             return null;
         }
     }

